src/main/java/org/isf/usergroups/rest/UserGroupController.java

```diff
@@ -0,0 +1,265 @@
+import org.isf.utils.exception.OHDataValidationException;
+import org.isf.utils.exception.OHServiceException;
+import org.isf.utils.exception.model.OHExceptionMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+import io.swagger.v3.oas.annotations.security.SecurityRequirement;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+
+@RestController(value = "/usergroups")
+@Tag(name = "User Groups")
+@SecurityRequirement(name = "bearerAuth")
+public class UserGroupController {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(UserGroupController.class);
+
+	@Autowired
+	private UserGroupMapper userGroupMapper;
+
+	@Autowired
+	private PermissionMapper permissionMapper;
+
+	@Autowired
+	private UserBrowsingManager userManager;
+
+	@Autowired
+	protected PermissionManager permissionManager;
+	@Autowired
+	protected GroupPermissionManager groupPermissionManager;
+
+	/**
+	 * Returns the list of {@link UserGroup}s.
+	 *
+	 * @return the list of {@link UserGroup}s
+	 */
+	@GetMapping(value = "/usergroups", produces = MediaType.APPLICATION_JSON_VALUE)
+	public List<UserGroupDTO> getUserGroups() throws OHServiceException {
+		LOGGER.info("Attempting to fetch the list of user groups.");
+		List<UserGroup> groups = userManager.getUserGroup();
+		List<UserGroupDTO> mappedGroups = userGroupMapper.map2DTOList(groups);
+		LOGGER.info("Found {} group(s).", mappedGroups.size());
+		return mappedGroups;
+	}
+
+	/**
+	 * Deletes a {@link UserGroup}.
+	 *
+	 * @param code - the code of the {@link UserGroup} to delete
+	 * @return {@code true} if the group has been deleted, {@code false} otherwise.
+	 */
+	@DeleteMapping(value = "/usergroups/{group_code}", produces = MediaType.APPLICATION_JSON_VALUE)
+	public Boolean deleteGroup(@PathVariable("group_code") String code) throws OHServiceException {
+		try {
+			UserGroup group = loadUserGroup(code);
+			userManager.deleteGroup(group);
+			return true;
+		} catch (OHServiceException serviceException) {
+			throw new OHAPIException(new OHExceptionMessage("User group not deleted."));
+		}
+	}
+
+	/**
+	 * Creates a new {@link UserGroup} with a minimum set of rights.
+	 *
+	 * @param userGroupDTO - the {@link UserGroup} to insert
+	 * @return {@code true} if the group has been inserted, {@code false} otherwise.
```

Comment:
where does it return false?   All I see is that on failure an exception is thrown